src/edu/ucsb/nceas/metacat/dataone/SystemMetadataFactory.java

```diff
@@ -658,20 +662,23 @@ public static SystemMetadata createSystemMetadata(boolean indexDataFile, String
 	 * Re-index the data file since the access rule was changed during the inserting of the eml document.
 	 * (During first time to index the data file in Metacat API, the eml hasn't been inserted)
 	 */
-	private static void reindexDataFile(Identifier id, SystemMetadata sysmeta) {
+	private static void reindexDataFile(Identifier id, SystemMetadata sysmeta, Session session, HttpServletRequest request) {
 	    try {
 	        logMetacat.debug("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ reindex"+id.getValue());
 	        if(sysmeta != null) {
-	            if(!sysmeta.getArchived()) {
-	                //set the archive to true to remove index.
-	                sysmeta.setArchived(true);
-	                MetacatSolrIndex.getInstance().submit(id, sysmeta, null, true);
-	                //re-insert the index
-	                sysmeta.setArchived(false);
-	                MetacatSolrIndex.getInstance().submit(id, sysmeta, null, true);
-	            } else {
-	                MetacatSolrIndex.getInstance().submit(id, sysmeta, null, true);
+	            for (int i=0; i<wait_times; i++) {
+	                try {
+	                    boolean exists = solrDocExists(id.getValue(), session, request);
+	                    if (exists) {
+	                        break;
+	                    }
+                        Thread.sleep(80);
+	                } catch (Exception ee) {
+	                    logMetacat.warn("Can't get the solr doc for  " + id.getValue() + " since " + ee.getMessage());
+	                    Thread.sleep(80);
```

Comment:
is this `index.resourcemap.waitingComponent.time` or should it be hardecoded?

Reply:
No, they are different. This is for morpho only. Since we will remove the support of Morpho soon, I decided not use new properties.